src/Analytics/Adapter.php

```diff
@@ -53,4 +64,129 @@ public function disable(): void
+
+        switch ($headers['Content-Type']) {
+            case 'application/json':
+                $query = json_encode($params);
+                break;
+
+            case 'multipart/form-data':
+                $query = $this->flatten($params);
+                break;
+
+            default:
+                $query = http_build_query($params);
+                break;
+        }
+
+        foreach ($headers as $i => $header) {
+            $headers[] = $i . ':' . $header;
+            unset($headers[$i]);
+        }
+
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_USERAGENT, php_uname('s') . '-' . php_uname('r') . ':php-' . phpversion());
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_HEADERFUNCTION, function($curl, $header) use (&$responseHeaders) {
+            $len = strlen($header);
+            $header = explode(':', strtolower($header), 2);
+
+            if (count($header) < 2) { // ignore invalid headers
+                return $len;
+            }
+
+            $responseHeaders[strtolower(trim($header[0]))] = trim($header[1]);
+
+            return $len;
+        });
+
+        if($method != 'GET') {
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $query);
+        }
+
+        $responseBody   = curl_exec($ch);
+
+        $responseType   = $responseHeaders['Content-Type'] ?? '';
+        $responseStatus = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        
+        switch(substr($responseType, 0, strpos($responseType, ';'))) {
+            case 'application/json':
+                $responseBody = json_decode($responseBody, true);
+            break;
+        }
+
+        if (curl_errno($ch)) {
+            throw new \Exception(curl_error($ch));
+        }
+        
+        curl_close($ch);
+
+        if($responseStatus >= 400) {
+            if(is_array($responseBody)) {
+                throw new \Exception(json_encode($responseBody));
+            } else {
+                throw new \Exception($responseStatus . ': ' . $responseBody);
+            }
+        }
+
+
+        return $responseBody;
+    }
+
+    /**
+     * Flatten params array to PHP multiple format
+     *
+     * @param array $data
+     * @param string $prefix
+     * @return array
+     */
+    protected function flatten(array $data, $prefix = '') {
+        $output = [];
+
+        foreach($data as $key => $value) {
+            $finalKey = $prefix ? "{$prefix}[{$key}]" : $key;
+
+            if (is_array($value)) {
+                $output += $this->flatten($value, $finalKey); // @todo: handle name collision here if needed
+            }
+            else {
+                $output[$finalKey] = $value;
+            }
+        }
+
+        return $output;
+    }
+
+    /**
+     * Log Error
+     */
+    protected function logError($e) {
```

Comment:
```suggestion
    protected function logError(Throwable $e) {
```
Is it a  Throwable or Exception ?